lib/recommendation-server.js

```diff
@@ -29,10 +29,13 @@ function RecommendationServer(opts) {
   this.xhr = opts.xhr;
   this.console = opts.console;
   this.prefs = opts.prefs;
+  this.resultClasses = opts.resultClasses;
   this.urlbarPrefs = opts.urlbarPrefs;
   this.server = this.prefs.getPrefType('server') ?
                 this.prefs.getCharPref('server') :
                 'https://universalsearch.testpilot.firefox.com';
+
+  this.getResultClass = this.getResultClass.bind(this);
```

Comment:
This line and `opts.resultClasses` / `this.resultClasses` should also go away when `getResultClass` is moved